In [1]:
import pandas as pd


In [27]:
df = pd.read_csv('../data/raw/phdb_dinosaur_occurrences.csv')
df.head()

C:\Users\john_\AppData\Local\Temp\ipykernel_23072\3316100536.py:1: DtypeWarning: Columns (56,81,82,83,84,119,131) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('../data/raw/phdb_dinosaur_occurrences.csv')


,occurrence_no,record_type,collection_no,identified_name,identified_rank,accepted_name,accepted_attr,accepted_rank,accepted_no,early_interval,...,max_ma_error,rare_body_parts,min_ma_error,direct_ma_value,direct_ma_unit,direct_ma_method,zone_type,direct_ma_error,plant_organ,artifacts
0,41524,occ,3257,Aves indet.,class,Aves,(Linnaeus 1758),class,36616,Lutetian,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,41580,occ,3256,Aves indet.,class,Aves,(Linnaeus 1758),class,36616,Ypresian,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,130209,occ,10755,Chaoyangosaurus liaosiensis n. gen. n. sp.,species,Chaoyangsaurus youngi,Zhao et al. 1999,species,55580,Late Kimmeridgian,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,130294,occ,10764,Protarchaeopteryx robusta n. gen. n. sp.,species,Protarchaeopteryx robusta,Ji and Ji 1997,species,66068,Late Barremian,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,130295,occ,10764,Caudipteryx zoui n. gen. n. sp.,species,Caudipteryx zoui,Ji et al. 1998,species,66066,Late Barremian,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [21]:
df.shape

(37817, 133)

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 37817 entries, 0 to 37816
Columns: 133 entries, occurrence_no to artifacts
dtypes: float64(17), int64(4), object(112)
memory usage: 38.4+ MB


In [28]:
df.columns.tolist()

['occurrence_no',
 'record_type',
 'collection_no',
 'identified_name',
 'identified_rank',
 'accepted_name',
 'accepted_attr',
 'accepted_rank',
 'accepted_no',
 'early_interval',
 'max_ma',
 'min_ma',
 'ref_author',
 'ref_pubyr',
 'reference_no',
 'phylum',
 'class',
 'lng',
 'lat',
 'occurrence_comments',
 'collection_name',
 'cc',
 'state',
 'county',
 'latlng_basis',
 'latlng_precision',
 'geogscale',
 'geogcomments',
 'paleomodel',
 'geoplate',
 'paleoage',
 'paleolng',
 'paleolat',
 'formation',
 'geological_group',
 'stratscale',
 'stratcomments',
 'lithology1',
 'environment',
 'size_classes',
 'pres_mode',
 'collection_type',
 'research_group',
 'taxon_environment',
 'motility',
 'life_habit',
 'diet',
 'reproduction',
 'ontogeny',
 'composition',
 'architecture',
 'identified_no',
 'late_interval',
 'order',
 'family',
 'genus',
 'abund_value',
 'abund_unit',
 'collection_aka',
 'altitude_value',
 'altitude_unit',
 'lithdescript',
 'lithadj1',
 'fossilsfrom1',
 'preservation

In [23]:
column_profile = pd.DataFrame({
    "column": df.columns,
    "dtype": df.dtypes.astype(str),
    "missing": df.isna().sum(),
    "missing_pct": (df.isna().mean() * 100).round(2),
    "unique": df.nunique()
})

column_profile

,column,dtype,missing,missing_pct,unique
occurrence_no,occurrence_no,int64,0,0.00,37817
record_type,record_type,object,0,0.00,1
collection_no,collection_no,int64,0,0.00,14351
identified_name,identified_name,object,0,0.00,10505
identified_rank,identified_rank,object,0,0.00,15
...,...,...,...,...,...
direct_ma_method,direct_ma_method,object,36837,97.41,11
zone_type,zone_type,object,36605,96.80,8
direct_ma_error,direct_ma_error,float64,36930,97.65,76
plant_organ,plant_organ,object,37809,99.98,2


In [24]:
column_profile.sort_values("missing_pct", ascending=False).head(30)

,column,dtype,missing,missing_pct,unique
plant_organ,plant_organ,object,37809,99.98,2
regionalbedunit,regionalbedunit,object,37746,99.81,1
regionalorder,regionalorder,object,37665,99.60,2
regionalbed,regionalbed,object,37594,99.41,73
ecospace_comments,ecospace_comments,object,37590,99.40,7
regionalsection,regionalsection,object,37564,99.33,30
container_no,container_no,float64,37464,99.07,30
rare_body_parts,rare_body_parts,object,37324,98.70,34
lithification2,lithification2,object,37304,98.64,3
vision,vision,object,37212,98.40,1


In [25]:
column_profile.sort_values("missing_pct").head(30)

,column,dtype,missing,missing_pct,unique
occurrence_no,occurrence_no,int64,0,0.00,37817
record_type,record_type,object,0,0.00,1
collection_no,collection_no,int64,0,0.00,14351
identified_name,identified_name,object,0,0.00,10505
identified_rank,identified_rank,object,0,0.00,15
accepted_name,accepted_name,object,0,0.00,6172
accepted_rank,accepted_rank,object,0,0.00,15
accepted_no,accepted_no,int64,0,0.00,6175
ref_author,ref_author,object,1,0.00,3818
early_interval,early_interval,object,0,0.00,253


In [26]:
column_profile[
    (column_profile["missing_pct"] >= 20) &
    (column_profile["missing_pct"] <= 80)
].sort_values("missing_pct")

,column,dtype,missing,missing_pct,unique
geogcomments,geogcomments,object,7883,20.85,9925
stratscale,stratscale,object,8381,22.16,5
family,family,object,8733,23.09,409
paleolat,paleolat,float64,11704,30.95,4801
paleolng,paleolng,float64,11704,30.95,6009
collection_methods,collection_methods,object,12016,31.77,389
genus,genus,object,14244,37.67,3087
formation,formation,object,14328,37.89,1778
stratcomments,stratcomments,object,15714,41.55,5746
abund_unit,abund_unit,object,15919,42.09,8


#Initial Data Quality Observations

23 columns of 131 are 100% filled with severaly columns missing 50% or more entires. The entirty of the bottom 30 of percentages of completed data are above 90% meaning potentially optional columns not filled for almost all fossil occurences. 